{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "import pandas as pd\n",
    "import rpy2.rinterface\n",
    "import numpy as np\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TGCA Survival Analysis of Pancreatic Cancer Patients\n",
    "\n",
    "### Modified from similar workflow described on biostars.org: https://www.biostars.org/p/153013/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download raw counts from Xena, place tsv in data folder\n",
    "# https://xenabrowser.net/datapages/?dataset=TCGA-PAAD%2FXena_Matrices%2FTCGA-PAAD.htseq_counts.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "HTSeq_raw_df = pd.read_csv(\"data/TCGA-PAAD.htseq_counts.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>TCGA-2J-AAB1-01A</th>\n",
       "      <th>TCGA-2J-AAB4-01A</th>\n",
       "      <th>TCGA-2J-AAB6-01A</th>\n",
       "      <th>TCGA-2J-AAB8-01A</th>\n",
       "      <th>TCGA-2J-AAB9-01A</th>\n",
       "      <th>TCGA-2J-AABA-01A</th>\n",
       "      <th>TCGA-2J-AABE-01A</th>\n",
       "      <th>TCGA-2J-AABF-01A</th>\n",
       "      <th>TCGA-2J-AABH-01A</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-XD-AAUH-01A</th>\n",
       "      <th>TCGA-XD-AAUI-01A</th>\n",
       "      <th>TCGA-XD-AAUL-01A</th>\n",
       "      <th>TCGA-XN-A8T3-01A</th>\n",
       "      <th>TCGA-XN-A8T5-01A</th>\n",
       "      <th>TCGA-YB-A89D-01A</th>\n",
       "      <th>TCGA-YB-A89D-11A</th>\n",
       "      <th>TCGA-YH-A8SY-01A</th>\n",
       "      <th>TCGA-YY-A8LH-01A</th>\n",
       "      <th>TCGA-Z5-AAPL-01A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.13</td>\n",
       "      <td>10.342075</td>\n",
       "      <td>10.997179</td>\n",
       "      <td>10.145932</td>\n",
       "      <td>9.424166</td>\n",
       "      <td>9.980140</td>\n",
       "      <td>10.856426</td>\n",
       "      <td>10.577429</td>\n",
       "      <td>11.116994</td>\n",
       "      <td>10.512740</td>\n",
       "      <td>...</td>\n",
       "      <td>10.290019</td>\n",
       "      <td>10.931476</td>\n",
       "      <td>11.259743</td>\n",
       "      <td>11.397140</td>\n",
       "      <td>10.302639</td>\n",
       "      <td>10.436712</td>\n",
       "      <td>10.644758</td>\n",
       "      <td>10.324181</td>\n",
       "      <td>10.901621</td>\n",
       "      <td>9.390169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.584963</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.584963</td>\n",
       "      <td>...</td>\n",
       "      <td>3.807355</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.584963</td>\n",
       "      <td>1.584963</td>\n",
       "      <td>8.383704</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.022368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.11</td>\n",
       "      <td>9.974415</td>\n",
       "      <td>10.329796</td>\n",
       "      <td>10.976564</td>\n",
       "      <td>9.531381</td>\n",
       "      <td>9.426265</td>\n",
       "      <td>9.930737</td>\n",
       "      <td>9.681238</td>\n",
       "      <td>10.680360</td>\n",
       "      <td>10.299208</td>\n",
       "      <td>...</td>\n",
       "      <td>10.104599</td>\n",
       "      <td>10.529431</td>\n",
       "      <td>10.484823</td>\n",
       "      <td>10.139551</td>\n",
       "      <td>10.225207</td>\n",
       "      <td>10.391244</td>\n",
       "      <td>10.267957</td>\n",
       "      <td>10.307201</td>\n",
       "      <td>10.611025</td>\n",
       "      <td>10.604553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.12</td>\n",
       "      <td>9.727920</td>\n",
       "      <td>9.751544</td>\n",
       "      <td>8.651052</td>\n",
       "      <td>9.398744</td>\n",
       "      <td>8.451211</td>\n",
       "      <td>9.231221</td>\n",
       "      <td>10.432542</td>\n",
       "      <td>10.520619</td>\n",
       "      <td>10.328675</td>\n",
       "      <td>...</td>\n",
       "      <td>9.497852</td>\n",
       "      <td>9.531381</td>\n",
       "      <td>9.463524</td>\n",
       "      <td>9.672425</td>\n",
       "      <td>9.002815</td>\n",
       "      <td>8.832890</td>\n",
       "      <td>8.731319</td>\n",
       "      <td>8.854868</td>\n",
       "      <td>9.868823</td>\n",
       "      <td>9.579316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.15</td>\n",
       "      <td>7.734710</td>\n",
       "      <td>7.942515</td>\n",
       "      <td>8.661778</td>\n",
       "      <td>7.672425</td>\n",
       "      <td>6.584963</td>\n",
       "      <td>7.383704</td>\n",
       "      <td>7.247928</td>\n",
       "      <td>9.074141</td>\n",
       "      <td>8.055282</td>\n",
       "      <td>...</td>\n",
       "      <td>7.159871</td>\n",
       "      <td>8.405141</td>\n",
       "      <td>7.900867</td>\n",
       "      <td>8.134426</td>\n",
       "      <td>7.159871</td>\n",
       "      <td>7.499846</td>\n",
       "      <td>7.294621</td>\n",
       "      <td>7.832890</td>\n",
       "      <td>8.044394</td>\n",
       "      <td>8.134426</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 183 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Ensembl_ID  TCGA-2J-AAB1-01A  TCGA-2J-AAB4-01A  TCGA-2J-AAB6-01A  \\\n",
       "0  ENSG00000000003.13         10.342075         10.997179         10.145932   \n",
       "1   ENSG00000000005.5          0.000000          0.000000          0.000000   \n",
       "2  ENSG00000000419.11          9.974415         10.329796         10.976564   \n",
       "3  ENSG00000000457.12          9.727920          9.751544          8.651052   \n",
       "4  ENSG00000000460.15          7.734710          7.942515          8.661778   \n",
       "\n",
       "   TCGA-2J-AAB8-01A  TCGA-2J-AAB9-01A  TCGA-2J-AABA-01A  TCGA-2J-AABE-01A  \\\n",
       "0          9.424166          9.980140         10.856426         10.577429   \n",
       "1          2.584963          0.000000          0.000000          1.000000   \n",
       "2          9.531381          9.426265          9.930737          9.681238   \n",
       "3          9.398744          8.451211          9.231221         10.432542   \n",
       "4          7.672425          6.584963          7.383704          7.247928   \n",
       "\n",
       "   TCGA-2J-AABF-01A  TCGA-2J-AABH-01A  ...  TCGA-XD-AAUH-01A  \\\n",
       "0         11.116994         10.512740  ...         10.290019   \n",
       "1          2.000000          1.584963  ...          3.807355   \n",
       "2         10.680360         10.299208  ...         10.104599   \n",
       "3         10.520619         10.328675  ...          9.497852   \n",
       "4          9.074141          8.055282  ...          7.159871   \n",
       "\n",
       "   TCGA-XD-AAUI-01A  TCGA-XD-AAUL-01A  TCGA-XN-A8T3-01A  TCGA-XN-A8T5-01A  \\\n",
       "0         10.931476         11.259743         11.397140         10.302639   \n",
       "1          1.000000          1.000000          2.000000          1.584963   \n",
       "2         10.529431         10.484823         10.139551         10.225207   \n",
       "3          9.531381          9.463524          9.672425          9.002815   \n",
       "4          8.405141          7.900867          8.134426          7.159871   \n",
       "\n",
       "   TCGA-YB-A89D-01A  TCGA-YB-A89D-11A  TCGA-YH-A8SY-01A  TCGA-YY-A8LH-01A  \\\n",
       "0         10.436712         10.644758         10.324181         10.901621   \n",
       "1          1.584963          8.383704          0.000000          0.000000   \n",
       "2         10.391244         10.267957         10.307201         10.611025   \n",
       "3          8.832890          8.731319          8.854868          9.868823   \n",
       "4          7.499846          7.294621          7.832890          8.044394   \n",
       "\n",
       "   TCGA-Z5-AAPL-01A  \n",
       "0          9.390169  \n",
       "1          6.022368  \n",
       "2         10.604553  \n",
       "3          9.579316  \n",
       "4          8.134426  \n",
       "\n",
       "[5 rows x 183 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize the head of the file\n",
    "HTSeq_raw_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These counts are actually partially processed (incremented by one then log transformed)\n",
    "# We want to undo this so that we are working with the actual raw counts\n",
    "HTSeq_raw_counts_df = pd.DataFrame()\n",
    "HTSeq_raw_counts_df['Ensembl_ID'] = HTSeq_raw_df['Ensembl_ID']\n",
    "for column in HTSeq_raw_df.columns.values[1:]:\n",
    "    series = HTSeq_raw_df[column]\n",
    "    series_antilog = (2**series) - 1\n",
    "    HTSeq_raw_counts_df[column] = series_antilog\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>TCGA-2J-AAB1-01A</th>\n",
       "      <th>TCGA-2J-AAB4-01A</th>\n",
       "      <th>TCGA-2J-AAB6-01A</th>\n",
       "      <th>TCGA-2J-AAB8-01A</th>\n",
       "      <th>TCGA-2J-AAB9-01A</th>\n",
       "      <th>TCGA-2J-AABA-01A</th>\n",
       "      <th>TCGA-2J-AABE-01A</th>\n",
       "      <th>TCGA-2J-AABF-01A</th>\n",
       "      <th>TCGA-2J-AABH-01A</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-XD-AAUH-01A</th>\n",
       "      <th>TCGA-XD-AAUI-01A</th>\n",
       "      <th>TCGA-XD-AAUL-01A</th>\n",
       "      <th>TCGA-XN-A8T3-01A</th>\n",
       "      <th>TCGA-XN-A8T5-01A</th>\n",
       "      <th>TCGA-YB-A89D-01A</th>\n",
       "      <th>TCGA-YB-A89D-11A</th>\n",
       "      <th>TCGA-YH-A8SY-01A</th>\n",
       "      <th>TCGA-YY-A8LH-01A</th>\n",
       "      <th>TCGA-Z5-AAPL-01A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.13</td>\n",
       "      <td>1297.0</td>\n",
       "      <td>2043.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>686.0</td>\n",
       "      <td>1009.0</td>\n",
       "      <td>1853.0</td>\n",
       "      <td>1527.0</td>\n",
       "      <td>2220.0</td>\n",
       "      <td>1460.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1251.0</td>\n",
       "      <td>1952.0</td>\n",
       "      <td>2451.0</td>\n",
       "      <td>2696.0</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>1385.0</td>\n",
       "      <td>1600.0</td>\n",
       "      <td>1281.0</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>670.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>333.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.11</td>\n",
       "      <td>1005.0</td>\n",
       "      <td>1286.0</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>739.0</td>\n",
       "      <td>687.0</td>\n",
       "      <td>975.0</td>\n",
       "      <td>820.0</td>\n",
       "      <td>1640.0</td>\n",
       "      <td>1259.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1100.0</td>\n",
       "      <td>1477.0</td>\n",
       "      <td>1432.0</td>\n",
       "      <td>1127.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>1342.0</td>\n",
       "      <td>1232.0</td>\n",
       "      <td>1266.0</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>1556.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.12</td>\n",
       "      <td>847.0</td>\n",
       "      <td>861.0</td>\n",
       "      <td>401.0</td>\n",
       "      <td>674.0</td>\n",
       "      <td>349.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>1381.0</td>\n",
       "      <td>1468.0</td>\n",
       "      <td>1285.0</td>\n",
       "      <td>...</td>\n",
       "      <td>722.0</td>\n",
       "      <td>739.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>815.0</td>\n",
       "      <td>512.0</td>\n",
       "      <td>455.0</td>\n",
       "      <td>424.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>934.0</td>\n",
       "      <td>764.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.15</td>\n",
       "      <td>212.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>538.0</td>\n",
       "      <td>265.0</td>\n",
       "      <td>...</td>\n",
       "      <td>142.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>263.0</td>\n",
       "      <td>280.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 183 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Ensembl_ID  TCGA-2J-AAB1-01A  TCGA-2J-AAB4-01A  TCGA-2J-AAB6-01A  \\\n",
       "0  ENSG00000000003.13            1297.0            2043.0            1132.0   \n",
       "1   ENSG00000000005.5               0.0               0.0               0.0   \n",
       "2  ENSG00000000419.11            1005.0            1286.0            2014.0   \n",
       "3  ENSG00000000457.12             847.0             861.0             401.0   \n",
       "4  ENSG00000000460.15             212.0             245.0             404.0   \n",
       "\n",
       "   TCGA-2J-AAB8-01A  TCGA-2J-AAB9-01A  TCGA-2J-AABA-01A  TCGA-2J-AABE-01A  \\\n",
       "0             686.0            1009.0            1853.0            1527.0   \n",
       "1               5.0               0.0               0.0               1.0   \n",
       "2             739.0             687.0             975.0             820.0   \n",
       "3             674.0             349.0             600.0            1381.0   \n",
       "4             203.0              95.0             166.0             151.0   \n",
       "\n",
       "   TCGA-2J-AABF-01A  TCGA-2J-AABH-01A  ...  TCGA-XD-AAUH-01A  \\\n",
       "0            2220.0            1460.0  ...            1251.0   \n",
       "1               3.0               2.0  ...              13.0   \n",
       "2            1640.0            1259.0  ...            1100.0   \n",
       "3            1468.0            1285.0  ...             722.0   \n",
       "4             538.0             265.0  ...             142.0   \n",
       "\n",
       "   TCGA-XD-AAUI-01A  TCGA-XD-AAUL-01A  TCGA-XN-A8T3-01A  TCGA-XN-A8T5-01A  \\\n",
       "0            1952.0            2451.0            2696.0            1262.0   \n",
       "1               1.0               1.0               3.0               2.0   \n",
       "2            1477.0            1432.0            1127.0            1196.0   \n",
       "3             739.0             705.0             815.0             512.0   \n",
       "4             338.0             238.0             280.0             142.0   \n",
       "\n",
       "   TCGA-YB-A89D-01A  TCGA-YB-A89D-11A  TCGA-YH-A8SY-01A  TCGA-YY-A8LH-01A  \\\n",
       "0            1385.0            1600.0            1281.0            1912.0   \n",
       "1               2.0             333.0               0.0               0.0   \n",
       "2            1342.0            1232.0            1266.0            1563.0   \n",
       "3             455.0             424.0             462.0             934.0   \n",
       "4             180.0             156.0             227.0             263.0   \n",
       "\n",
       "   TCGA-Z5-AAPL-01A  \n",
       "0             670.0  \n",
       "1              64.0  \n",
       "2            1556.0  \n",
       "3             764.0  \n",
       "4             280.0  \n",
       "\n",
       "[5 rows x 183 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize the raw counts\n",
    "HTSeq_raw_counts_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to figure out which columns are tumor samples and which columns are normal samples\n",
    "origin_dict = {}\n",
    "for column in list(HTSeq_raw_counts_df)[1:]:\n",
    "    if column[13] == '0':\n",
    "        origin_dict[column] = 'tumor'\n",
    "    elif column[13] == '1':\n",
    "        origin_dict[column] = 'normal'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'tumor': 178, 'normal': 4})"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of occurances of each tumor or normal tissue\n",
    "Counter(origin_dict.values())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For this analysis I am only interested in tumor samples\n",
    "# I subset the HTSeq_raw_counts_df to just get tumor samples\n",
    "slice_list = ['Ensembl_ID']\n",
    "for column, value in origin_dict.items():\n",
    "    if value == 'tumor':\n",
    "        slice_list.append(column)\n",
    "        \n",
    "HTSeq_raw_counts_tumors_df = HTSeq_raw_counts_df[slice_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>TCGA-2J-AAB1-01A</th>\n",
       "      <th>TCGA-2J-AAB4-01A</th>\n",
       "      <th>TCGA-2J-AAB6-01A</th>\n",
       "      <th>TCGA-2J-AAB8-01A</th>\n",
       "      <th>TCGA-2J-AAB9-01A</th>\n",
       "      <th>TCGA-2J-AABA-01A</th>\n",
       "      <th>TCGA-2J-AABE-01A</th>\n",
       "      <th>TCGA-2J-AABF-01A</th>\n",
       "      <th>TCGA-2J-AABH-01A</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-XD-AAUG-01A</th>\n",
       "      <th>TCGA-XD-AAUH-01A</th>\n",
       "      <th>TCGA-XD-AAUI-01A</th>\n",
       "      <th>TCGA-XD-AAUL-01A</th>\n",
       "      <th>TCGA-XN-A8T3-01A</th>\n",
       "      <th>TCGA-XN-A8T5-01A</th>\n",
       "      <th>TCGA-YB-A89D-01A</th>\n",
       "      <th>TCGA-YH-A8SY-01A</th>\n",
       "      <th>TCGA-YY-A8LH-01A</th>\n",
       "      <th>TCGA-Z5-AAPL-01A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.13</td>\n",
       "      <td>1297.0</td>\n",
       "      <td>2043.0</td>\n",
       "      <td>1132.0</td>\n",
       "      <td>686.0</td>\n",
       "      <td>1009.0</td>\n",
       "      <td>1853.0</td>\n",
       "      <td>1527.0</td>\n",
       "      <td>2220.0</td>\n",
       "      <td>1460.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1126.0</td>\n",
       "      <td>1251.0</td>\n",
       "      <td>1952.0</td>\n",
       "      <td>2451.0</td>\n",
       "      <td>2696.0</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>1385.0</td>\n",
       "      <td>1281.0</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>670.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>33.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.11</td>\n",
       "      <td>1005.0</td>\n",
       "      <td>1286.0</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>739.0</td>\n",
       "      <td>687.0</td>\n",
       "      <td>975.0</td>\n",
       "      <td>820.0</td>\n",
       "      <td>1640.0</td>\n",
       "      <td>1259.0</td>\n",
       "      <td>...</td>\n",
       "      <td>835.0</td>\n",
       "      <td>1100.0</td>\n",
       "      <td>1477.0</td>\n",
       "      <td>1432.0</td>\n",
       "      <td>1127.0</td>\n",
       "      <td>1196.0</td>\n",
       "      <td>1342.0</td>\n",
       "      <td>1266.0</td>\n",
       "      <td>1563.0</td>\n",
       "      <td>1556.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.12</td>\n",
       "      <td>847.0</td>\n",
       "      <td>861.0</td>\n",
       "      <td>401.0</td>\n",
       "      <td>674.0</td>\n",
       "      <td>349.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>1381.0</td>\n",
       "      <td>1468.0</td>\n",
       "      <td>1285.0</td>\n",
       "      <td>...</td>\n",
       "      <td>462.0</td>\n",
       "      <td>722.0</td>\n",
       "      <td>739.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>815.0</td>\n",
       "      <td>512.0</td>\n",
       "      <td>455.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>934.0</td>\n",
       "      <td>764.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.15</td>\n",
       "      <td>212.0</td>\n",
       "      <td>245.0</td>\n",
       "      <td>404.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>538.0</td>\n",
       "      <td>265.0</td>\n",
       "      <td>...</td>\n",
       "      <td>106.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>238.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>142.0</td>\n",
       "      <td>180.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>263.0</td>\n",
       "      <td>280.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Ensembl_ID  TCGA-2J-AAB1-01A  TCGA-2J-AAB4-01A  TCGA-2J-AAB6-01A  \\\n",
       "0  ENSG00000000003.13            1297.0            2043.0            1132.0   \n",
       "1   ENSG00000000005.5               0.0               0.0               0.0   \n",
       "2  ENSG00000000419.11            1005.0            1286.0            2014.0   \n",
       "3  ENSG00000000457.12             847.0             861.0             401.0   \n",
       "4  ENSG00000000460.15             212.0             245.0             404.0   \n",
       "\n",
       "   TCGA-2J-AAB8-01A  TCGA-2J-AAB9-01A  TCGA-2J-AABA-01A  TCGA-2J-AABE-01A  \\\n",
       "0             686.0            1009.0            1853.0            1527.0   \n",
       "1               5.0               0.0               0.0               1.0   \n",
       "2             739.0             687.0             975.0             820.0   \n",
       "3             674.0             349.0             600.0            1381.0   \n",
       "4             203.0              95.0             166.0             151.0   \n",
       "\n",
       "   TCGA-2J-AABF-01A  TCGA-2J-AABH-01A  ...  TCGA-XD-AAUG-01A  \\\n",
       "0            2220.0            1460.0  ...            1126.0   \n",
       "1               3.0               2.0  ...              33.0   \n",
       "2            1640.0            1259.0  ...             835.0   \n",
       "3            1468.0            1285.0  ...             462.0   \n",
       "4             538.0             265.0  ...             106.0   \n",
       "\n",
       "   TCGA-XD-AAUH-01A  TCGA-XD-AAUI-01A  TCGA-XD-AAUL-01A  TCGA-XN-A8T3-01A  \\\n",
       "0            1251.0            1952.0            2451.0            2696.0   \n",
       "1              13.0               1.0               1.0               3.0   \n",
       "2            1100.0            1477.0            1432.0            1127.0   \n",
       "3             722.0             739.0             705.0             815.0   \n",
       "4             142.0             338.0             238.0             280.0   \n",
       "\n",
       "   TCGA-XN-A8T5-01A  TCGA-YB-A89D-01A  TCGA-YH-A8SY-01A  TCGA-YY-A8LH-01A  \\\n",
       "0            1262.0            1385.0            1281.0            1912.0   \n",
       "1               2.0               2.0               0.0               0.0   \n",
       "2            1196.0            1342.0            1266.0            1563.0   \n",
       "3             512.0             455.0             462.0             934.0   \n",
       "4             142.0             180.0             227.0             263.0   \n",
       "\n",
       "   TCGA-Z5-AAPL-01A  \n",
       "0             670.0  \n",
       "1              64.0  \n",
       "2            1556.0  \n",
       "3             764.0  \n",
       "4             280.0  \n",
       "\n",
       "[5 rows x 179 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize the HTSeq_raw_counts_tumors_df\n",
    "HTSeq_raw_counts_tumors_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Probably a good idea to remove lowly expressed genes\n",
    "# Find out where genes have expression of 0 if 50% of samples\n",
    "zeroes = (HTSeq_raw_counts_tumors_df == 0).astype(int)\n",
    "zeroes_count = zeroes.sum(axis=1).to_frame()\n",
    "length = zeroes.shape[1]\n",
    "zeroes_count = zeroes_count.drop(zeroes_count[zeroes_count[0] < length/2].index)\n",
    "drop_index_list = zeroes_count.index.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Drop rows where genes have expression of 0 if 50% of samples\n",
    "HTSeq_raw_counts_tumors_no_zeroes_df = HTSeq_raw_counts_tumors_df.drop(labels=drop_index_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29034"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Determine how many rows were dropped due to low gene expression\n",
    "before_drop = HTSeq_raw_counts_tumors_df.shape[0]\n",
    "after_drop = HTSeq_raw_counts_tumors_no_zeroes_df.shape[0]\n",
    "number_dropped = before_drop - after_drop\n",
    "number_dropped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to convert floats counts to integer, otherwise DESeq2 will complain\n",
    "for column in HTSeq_raw_counts_tumors_no_zeroes_df.columns.values[1:]:\n",
    "    HTSeq_raw_counts_tumors_no_zeroes_df[column] = HTSeq_raw_counts_tumors_no_zeroes_df[column].astype(int)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>TCGA-2J-AAB1-01A</th>\n",
       "      <th>TCGA-2J-AAB4-01A</th>\n",
       "      <th>TCGA-2J-AAB6-01A</th>\n",
       "      <th>TCGA-2J-AAB8-01A</th>\n",
       "      <th>TCGA-2J-AAB9-01A</th>\n",
       "      <th>TCGA-2J-AABA-01A</th>\n",
       "      <th>TCGA-2J-AABE-01A</th>\n",
       "      <th>TCGA-2J-AABF-01A</th>\n",
       "      <th>TCGA-2J-AABH-01A</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-XD-AAUG-01A</th>\n",
       "      <th>TCGA-XD-AAUH-01A</th>\n",
       "      <th>TCGA-XD-AAUI-01A</th>\n",
       "      <th>TCGA-XD-AAUL-01A</th>\n",
       "      <th>TCGA-XN-A8T3-01A</th>\n",
       "      <th>TCGA-XN-A8T5-01A</th>\n",
       "      <th>TCGA-YB-A89D-01A</th>\n",
       "      <th>TCGA-YH-A8SY-01A</th>\n",
       "      <th>TCGA-YY-A8LH-01A</th>\n",
       "      <th>TCGA-Z5-AAPL-01A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003.13</td>\n",
       "      <td>1297</td>\n",
       "      <td>2042</td>\n",
       "      <td>1131</td>\n",
       "      <td>686</td>\n",
       "      <td>1009</td>\n",
       "      <td>1852</td>\n",
       "      <td>1526</td>\n",
       "      <td>2219</td>\n",
       "      <td>1459</td>\n",
       "      <td>...</td>\n",
       "      <td>1126</td>\n",
       "      <td>1250</td>\n",
       "      <td>1952</td>\n",
       "      <td>2451</td>\n",
       "      <td>2695</td>\n",
       "      <td>1262</td>\n",
       "      <td>1384</td>\n",
       "      <td>1280</td>\n",
       "      <td>1911</td>\n",
       "      <td>669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>32</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419.11</td>\n",
       "      <td>1005</td>\n",
       "      <td>1285</td>\n",
       "      <td>2013</td>\n",
       "      <td>739</td>\n",
       "      <td>686</td>\n",
       "      <td>974</td>\n",
       "      <td>820</td>\n",
       "      <td>1639</td>\n",
       "      <td>1259</td>\n",
       "      <td>...</td>\n",
       "      <td>834</td>\n",
       "      <td>1100</td>\n",
       "      <td>1476</td>\n",
       "      <td>1432</td>\n",
       "      <td>1127</td>\n",
       "      <td>1195</td>\n",
       "      <td>1341</td>\n",
       "      <td>1265</td>\n",
       "      <td>1562</td>\n",
       "      <td>1556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457.12</td>\n",
       "      <td>846</td>\n",
       "      <td>861</td>\n",
       "      <td>401</td>\n",
       "      <td>674</td>\n",
       "      <td>348</td>\n",
       "      <td>599</td>\n",
       "      <td>1381</td>\n",
       "      <td>1468</td>\n",
       "      <td>1284</td>\n",
       "      <td>...</td>\n",
       "      <td>461</td>\n",
       "      <td>721</td>\n",
       "      <td>739</td>\n",
       "      <td>704</td>\n",
       "      <td>814</td>\n",
       "      <td>512</td>\n",
       "      <td>454</td>\n",
       "      <td>461</td>\n",
       "      <td>933</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460.15</td>\n",
       "      <td>212</td>\n",
       "      <td>245</td>\n",
       "      <td>403</td>\n",
       "      <td>202</td>\n",
       "      <td>94</td>\n",
       "      <td>165</td>\n",
       "      <td>150</td>\n",
       "      <td>537</td>\n",
       "      <td>264</td>\n",
       "      <td>...</td>\n",
       "      <td>105</td>\n",
       "      <td>142</td>\n",
       "      <td>338</td>\n",
       "      <td>237</td>\n",
       "      <td>279</td>\n",
       "      <td>142</td>\n",
       "      <td>179</td>\n",
       "      <td>226</td>\n",
       "      <td>263</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Ensembl_ID  TCGA-2J-AAB1-01A  TCGA-2J-AAB4-01A  TCGA-2J-AAB6-01A  \\\n",
       "0  ENSG00000000003.13              1297              2042              1131   \n",
       "1   ENSG00000000005.5                 0                 0                 0   \n",
       "2  ENSG00000000419.11              1005              1285              2013   \n",
       "3  ENSG00000000457.12               846               861               401   \n",
       "4  ENSG00000000460.15               212               245               403   \n",
       "\n",
       "   TCGA-2J-AAB8-01A  TCGA-2J-AAB9-01A  TCGA-2J-AABA-01A  TCGA-2J-AABE-01A  \\\n",
       "0               686              1009              1852              1526   \n",
       "1                 4                 0                 0                 1   \n",
       "2               739               686               974               820   \n",
       "3               674               348               599              1381   \n",
       "4               202                94               165               150   \n",
       "\n",
       "   TCGA-2J-AABF-01A  TCGA-2J-AABH-01A  ...  TCGA-XD-AAUG-01A  \\\n",
       "0              2219              1459  ...              1126   \n",
       "1                 3                 1  ...                32   \n",
       "2              1639              1259  ...               834   \n",
       "3              1468              1284  ...               461   \n",
       "4               537               264  ...               105   \n",
       "\n",
       "   TCGA-XD-AAUH-01A  TCGA-XD-AAUI-01A  TCGA-XD-AAUL-01A  TCGA-XN-A8T3-01A  \\\n",
       "0              1250              1952              2451              2695   \n",
       "1                13                 1                 1                 3   \n",
       "2              1100              1476              1432              1127   \n",
       "3               721               739               704               814   \n",
       "4               142               338               237               279   \n",
       "\n",
       "   TCGA-XN-A8T5-01A  TCGA-YB-A89D-01A  TCGA-YH-A8SY-01A  TCGA-YY-A8LH-01A  \\\n",
       "0              1262              1384              1280              1911   \n",
       "1                 1                 1                 0                 0   \n",
       "2              1195              1341              1265              1562   \n",
       "3               512               454               461               933   \n",
       "4               142               179               226               263   \n",
       "\n",
       "   TCGA-Z5-AAPL-01A  \n",
       "0               669  \n",
       "1                64  \n",
       "2              1556  \n",
       "3               763  \n",
       "4               279  \n",
       "\n",
       "[5 rows x 179 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize results of float -> int conversion\n",
    "HTSeq_raw_counts_tumors_no_zeroes_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Iteratively save a tsv corresponing to the raw counts for each tumor\n",
    "# needed to pass data off to DESeq2\n",
    "for column in HTSeq_raw_counts_tumors_no_zeroes_df.columns.values[1:]:\n",
    "    file_name = 'data/tumor_samples/' + column + '.tsv'\n",
    "    HTSeq_raw_counts_tumors_no_zeroes_df[['Ensembl_ID', column]].to_csv(file_name, sep=\" \", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Loading required package: S4Vectors\n",
      "\n",
      "R[write to console]: Loading required package: stats4\n",
      "\n",
      "R[write to console]: Loading required package: BiocGenerics\n",
      "\n",
      "R[write to console]: Loading required package: parallel\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,\n",
      "    colnames, colSums, dirname, do.call, duplicated, eval, evalq,\n",
      "    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,\n",
      "    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,\n",
      "    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,\n",
      "    rowSums, sapply, setdiff, sort, table, tapply, union, unique,\n",
      "    unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: IRanges\n",
      "\n",
      "R[write to console]: Loading required package: GenomicRanges\n",
      "\n",
      "R[write to console]: Loading required package: GenomeInfoDb\n",
      "\n",
      "R[write to console]: Loading required package: SummarizedExperiment\n",
      "\n",
      "R[write to console]: Loading required package: Biobase\n",
      "\n",
      "R[write to console]: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: DelayedArray\n",
      "\n",
      "R[write to console]: Loading required package: matrixStats\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: BiocParallel\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘DelayedArray’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    aperm, apply\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# load the R libraries\n",
    "library(\"DESeq2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "          sampleName             fileName condition\n",
       "1   TCGA-2J-AAB1-01A TCGA-2J-AAB1-01A.tsv     tumor\n",
       "2   TCGA-2J-AAB4-01A TCGA-2J-AAB4-01A.tsv     tumor\n",
       "3   TCGA-2J-AAB6-01A TCGA-2J-AAB6-01A.tsv     tumor\n",
       "4   TCGA-2J-AAB8-01A TCGA-2J-AAB8-01A.tsv     tumor\n",
       "5   TCGA-2J-AAB9-01A TCGA-2J-AAB9-01A.tsv     tumor\n",
       "6   TCGA-2J-AABA-01A TCGA-2J-AABA-01A.tsv     tumor\n",
       "7   TCGA-2J-AABE-01A TCGA-2J-AABE-01A.tsv     tumor\n",
       "8   TCGA-2J-AABF-01A TCGA-2J-AABF-01A.tsv     tumor\n",
       "9   TCGA-2J-AABH-01A TCGA-2J-AABH-01A.tsv     tumor\n",
       "10  TCGA-2J-AABI-01A TCGA-2J-AABI-01A.tsv     tumor\n",
       "11  TCGA-2J-AABK-01A TCGA-2J-AABK-01A.tsv     tumor\n",
       "12  TCGA-2J-AABO-01A TCGA-2J-AABO-01A.tsv     tumor\n",
       "13  TCGA-2J-AABP-01A TCGA-2J-AABP-01A.tsv     tumor\n",
       "14  TCGA-2J-AABR-01A TCGA-2J-AABR-01A.tsv     tumor\n",
       "15  TCGA-2J-AABT-01A TCGA-2J-AABT-01A.tsv     tumor\n",
       "16  TCGA-2J-AABU-01A TCGA-2J-AABU-01A.tsv     tumor\n",
       "17  TCGA-2J-AABV-01A TCGA-2J-AABV-01A.tsv     tumor\n",
       "18  TCGA-2L-AAQA-01A TCGA-2L-AAQA-01A.tsv     tumor\n",
       "19  TCGA-2L-AAQE-01A TCGA-2L-AAQE-01A.tsv     tumor\n",
       "20  TCGA-2L-AAQI-01A TCGA-2L-AAQI-01A.tsv     tumor\n",
       "21  TCGA-2L-AAQJ-01A TCGA-2L-AAQJ-01A.tsv     tumor\n",
       "22  TCGA-2L-AAQL-01A TCGA-2L-AAQL-01A.tsv     tumor\n",
       "23  TCGA-2L-AAQM-01A TCGA-2L-AAQM-01A.tsv     tumor\n",
       "24  TCGA-3A-A9I5-01A TCGA-3A-A9I5-01A.tsv     tumor\n",
       "25  TCGA-3A-A9I7-01A TCGA-3A-A9I7-01A.tsv     tumor\n",
       "26  TCGA-3A-A9I9-01A TCGA-3A-A9I9-01A.tsv     tumor\n",
       "27  TCGA-3A-A9IB-01A TCGA-3A-A9IB-01A.tsv     tumor\n",
       "28  TCGA-3A-A9IC-01A TCGA-3A-A9IC-01A.tsv     tumor\n",
       "29  TCGA-3A-A9IH-01A TCGA-3A-A9IH-01A.tsv     tumor\n",
       "30  TCGA-3A-A9IJ-01A TCGA-3A-A9IJ-01A.tsv     tumor\n",
       "31  TCGA-3A-A9IL-01A TCGA-3A-A9IL-01A.tsv     tumor\n",
       "32  TCGA-3A-A9IN-01A TCGA-3A-A9IN-01A.tsv     tumor\n",
       "33  TCGA-3A-A9IO-01A TCGA-3A-A9IO-01A.tsv     tumor\n",
       "34  TCGA-3A-A9IR-01A TCGA-3A-A9IR-01A.tsv     tumor\n",
       "35  TCGA-3A-A9IS-01A TCGA-3A-A9IS-01A.tsv     tumor\n",
       "36  TCGA-3A-A9IU-01A TCGA-3A-A9IU-01A.tsv     tumor\n",
       "37  TCGA-3A-A9IV-01A TCGA-3A-A9IV-01A.tsv     tumor\n",
       "38  TCGA-3A-A9IX-01A TCGA-3A-A9IX-01A.tsv     tumor\n",
       "39  TCGA-3A-A9IZ-01A TCGA-3A-A9IZ-01A.tsv     tumor\n",
       "40  TCGA-3A-A9J0-01A TCGA-3A-A9J0-01A.tsv     tumor\n",
       "41  TCGA-3E-AAAY-01A TCGA-3E-AAAY-01A.tsv     tumor\n",
       "42  TCGA-3E-AAAZ-01A TCGA-3E-AAAZ-01A.tsv     tumor\n",
       "43  TCGA-F2-6879-01A TCGA-F2-6879-01A.tsv     tumor\n",
       "44  TCGA-F2-6880-01A TCGA-F2-6880-01A.tsv     tumor\n",
       "45  TCGA-F2-7273-01A TCGA-F2-7273-01A.tsv     tumor\n",
       "46  TCGA-F2-7276-01A TCGA-F2-7276-01A.tsv     tumor\n",
       "47  TCGA-F2-A44G-01A TCGA-F2-A44G-01A.tsv     tumor\n",
       "48  TCGA-F2-A44H-01A TCGA-F2-A44H-01A.tsv     tumor\n",
       "49  TCGA-F2-A7TX-01A TCGA-F2-A7TX-01A.tsv     tumor\n",
       "50  TCGA-F2-A8YN-01A TCGA-F2-A8YN-01A.tsv     tumor\n",
       "51  TCGA-FB-A4P5-01A TCGA-FB-A4P5-01A.tsv     tumor\n",
       "52  TCGA-FB-A4P6-01A TCGA-FB-A4P6-01A.tsv     tumor\n",
       "53  TCGA-FB-A545-01A TCGA-FB-A545-01A.tsv     tumor\n",
       "54  TCGA-FB-A5VM-01A TCGA-FB-A5VM-01A.tsv     tumor\n",
       "55  TCGA-FB-A78T-01A TCGA-FB-A78T-01A.tsv     tumor\n",
       "56  TCGA-FB-A7DR-01A TCGA-FB-A7DR-01A.tsv     tumor\n",
       "57  TCGA-FB-AAPP-01A TCGA-FB-AAPP-01A.tsv     tumor\n",
       "58  TCGA-FB-AAPQ-01A TCGA-FB-AAPQ-01A.tsv     tumor\n",
       "59  TCGA-FB-AAPS-01A TCGA-FB-AAPS-01A.tsv     tumor\n",
       "60  TCGA-FB-AAPU-01A TCGA-FB-AAPU-01A.tsv     tumor\n",
       "61  TCGA-FB-AAPY-01A TCGA-FB-AAPY-01A.tsv     tumor\n",
       "62  TCGA-FB-AAPZ-01A TCGA-FB-AAPZ-01A.tsv     tumor\n",
       "63  TCGA-FB-AAQ0-01A TCGA-FB-AAQ0-01A.tsv     tumor\n",
       "64  TCGA-FB-AAQ1-01A TCGA-FB-AAQ1-01A.tsv     tumor\n",
       "65  TCGA-FB-AAQ2-01A TCGA-FB-AAQ2-01A.tsv     tumor\n",
       "66  TCGA-FB-AAQ3-01A TCGA-FB-AAQ3-01A.tsv     tumor\n",
       "67  TCGA-FB-AAQ6-01A TCGA-FB-AAQ6-01A.tsv     tumor\n",
       "68  TCGA-H6-8124-01A TCGA-H6-8124-01A.tsv     tumor\n",
       "69  TCGA-H6-A45N-01A TCGA-H6-A45N-01A.tsv     tumor\n",
       "70  TCGA-H8-A6C1-01A TCGA-H8-A6C1-01A.tsv     tumor\n",
       "71  TCGA-HV-A5A3-01A TCGA-HV-A5A3-01A.tsv     tumor\n",
       "72  TCGA-HV-A5A4-01A TCGA-HV-A5A4-01A.tsv     tumor\n",
       "73  TCGA-HV-A5A5-01A TCGA-HV-A5A5-01A.tsv     tumor\n",
       "74  TCGA-HV-A5A6-01A TCGA-HV-A5A6-01A.tsv     tumor\n",
       "75  TCGA-HV-A7OL-01A TCGA-HV-A7OL-01A.tsv     tumor\n",
       "76  TCGA-HV-A7OP-01A TCGA-HV-A7OP-01A.tsv     tumor\n",
       "77  TCGA-HV-AA8V-01A TCGA-HV-AA8V-01A.tsv     tumor\n",
       "78  TCGA-HV-AA8X-01A TCGA-HV-AA8X-01A.tsv     tumor\n",
       "79  TCGA-HZ-7289-01A TCGA-HZ-7289-01A.tsv     tumor\n",
       "80  TCGA-HZ-7918-01A TCGA-HZ-7918-01A.tsv     tumor\n",
       "81  TCGA-HZ-7919-01A TCGA-HZ-7919-01A.tsv     tumor\n",
       "82  TCGA-HZ-7920-01A TCGA-HZ-7920-01A.tsv     tumor\n",
       "83  TCGA-HZ-7922-01A TCGA-HZ-7922-01A.tsv     tumor\n",
       "84  TCGA-HZ-7923-01A TCGA-HZ-7923-01A.tsv     tumor\n",
       "85  TCGA-HZ-7924-01A TCGA-HZ-7924-01A.tsv     tumor\n",
       "86  TCGA-HZ-7925-01A TCGA-HZ-7925-01A.tsv     tumor\n",
       "87  TCGA-HZ-7926-01A TCGA-HZ-7926-01A.tsv     tumor\n",
       "88  TCGA-HZ-8001-01A TCGA-HZ-8001-01A.tsv     tumor\n",
       "89  TCGA-HZ-8002-01A TCGA-HZ-8002-01A.tsv     tumor\n",
       "90  TCGA-HZ-8003-01A TCGA-HZ-8003-01A.tsv     tumor\n",
       "91  TCGA-HZ-8005-01A TCGA-HZ-8005-01A.tsv     tumor\n",
       "92  TCGA-HZ-8315-01A TCGA-HZ-8315-01A.tsv     tumor\n",
       "93  TCGA-HZ-8317-01A TCGA-HZ-8317-01A.tsv     tumor\n",
       "94  TCGA-HZ-8519-01A TCGA-HZ-8519-01A.tsv     tumor\n",
       "95  TCGA-HZ-8636-01A TCGA-HZ-8636-01A.tsv     tumor\n",
       "96  TCGA-HZ-8637-01A TCGA-HZ-8637-01A.tsv     tumor\n",
       "97  TCGA-HZ-8638-01A TCGA-HZ-8638-01A.tsv     tumor\n",
       "98  TCGA-HZ-A49G-01A TCGA-HZ-A49G-01A.tsv     tumor\n",
       "99  TCGA-HZ-A49H-01A TCGA-HZ-A49H-01A.tsv     tumor\n",
       "100 TCGA-HZ-A49I-01A TCGA-HZ-A49I-01A.tsv     tumor\n",
       "101 TCGA-HZ-A4BH-01A TCGA-HZ-A4BH-01A.tsv     tumor\n",
       "102 TCGA-HZ-A4BK-01A TCGA-HZ-A4BK-01A.tsv     tumor\n",
       "103 TCGA-HZ-A77O-01A TCGA-HZ-A77O-01A.tsv     tumor\n",
       "104 TCGA-HZ-A77P-01A TCGA-HZ-A77P-01A.tsv     tumor\n",
       "105 TCGA-HZ-A77Q-01A TCGA-HZ-A77Q-01A.tsv     tumor\n",
       "106 TCGA-HZ-A8P0-01A TCGA-HZ-A8P0-01A.tsv     tumor\n",
       "107 TCGA-HZ-A8P1-01A TCGA-HZ-A8P1-01A.tsv     tumor\n",
       "108 TCGA-HZ-A9TJ-01A TCGA-HZ-A9TJ-01A.tsv     tumor\n",
       "109 TCGA-HZ-A9TJ-06A TCGA-HZ-A9TJ-06A.tsv     tumor\n",
       "110 TCGA-IB-7644-01A TCGA-IB-7644-01A.tsv     tumor\n",
       "111 TCGA-IB-7645-01A TCGA-IB-7645-01A.tsv     tumor\n",
       "112 TCGA-IB-7646-01A TCGA-IB-7646-01A.tsv     tumor\n",
       "113 TCGA-IB-7649-01A TCGA-IB-7649-01A.tsv     tumor\n",
       "114 TCGA-IB-7651-01A TCGA-IB-7651-01A.tsv     tumor\n",
       "115 TCGA-IB-7652-01A TCGA-IB-7652-01A.tsv     tumor\n",
       "116 TCGA-IB-7654-01A TCGA-IB-7654-01A.tsv     tumor\n",
       "117 TCGA-IB-7885-01A TCGA-IB-7885-01A.tsv     tumor\n",
       "118 TCGA-IB-7886-01A TCGA-IB-7886-01A.tsv     tumor\n",
       "119 TCGA-IB-7887-01A TCGA-IB-7887-01A.tsv     tumor\n",
       "120 TCGA-IB-7888-01A TCGA-IB-7888-01A.tsv     tumor\n",
       "121 TCGA-IB-7889-01A TCGA-IB-7889-01A.tsv     tumor\n",
       "122 TCGA-IB-7890-01A TCGA-IB-7890-01A.tsv     tumor\n",
       "123 TCGA-IB-7891-01A TCGA-IB-7891-01A.tsv     tumor\n",
       "124 TCGA-IB-7893-01A TCGA-IB-7893-01A.tsv     tumor\n",
       "125 TCGA-IB-7897-01A TCGA-IB-7897-01A.tsv     tumor\n",
       "126 TCGA-IB-8126-01A TCGA-IB-8126-01A.tsv     tumor\n",
       "127 TCGA-IB-8127-01A TCGA-IB-8127-01A.tsv     tumor\n",
       "128 TCGA-IB-A5SO-01A TCGA-IB-A5SO-01A.tsv     tumor\n",
       "129 TCGA-IB-A5SP-01A TCGA-IB-A5SP-01A.tsv     tumor\n",
       "130 TCGA-IB-A5SQ-01A TCGA-IB-A5SQ-01A.tsv     tumor\n",
       "131 TCGA-IB-A5SS-01A TCGA-IB-A5SS-01A.tsv     tumor\n",
       "132 TCGA-IB-A5ST-01A TCGA-IB-A5ST-01A.tsv     tumor\n",
       "133 TCGA-IB-A6UF-01A TCGA-IB-A6UF-01A.tsv     tumor\n",
       "134 TCGA-IB-A6UG-01A TCGA-IB-A6UG-01A.tsv     tumor\n",
       "135 TCGA-IB-A7LX-01A TCGA-IB-A7LX-01A.tsv     tumor\n",
       "136 TCGA-IB-A7M4-01A TCGA-IB-A7M4-01A.tsv     tumor\n",
       "137 TCGA-IB-AAUM-01A TCGA-IB-AAUM-01A.tsv     tumor\n",
       "138 TCGA-IB-AAUN-01A TCGA-IB-AAUN-01A.tsv     tumor\n",
       "139 TCGA-IB-AAUO-01A TCGA-IB-AAUO-01A.tsv     tumor\n",
       "140 TCGA-IB-AAUP-01A TCGA-IB-AAUP-01A.tsv     tumor\n",
       "141 TCGA-IB-AAUQ-01A TCGA-IB-AAUQ-01A.tsv     tumor\n",
       "142 TCGA-IB-AAUR-01A TCGA-IB-AAUR-01A.tsv     tumor\n",
       "143 TCGA-IB-AAUS-01A TCGA-IB-AAUS-01A.tsv     tumor\n",
       "144 TCGA-IB-AAUT-01A TCGA-IB-AAUT-01A.tsv     tumor\n",
       "145 TCGA-IB-AAUU-01A TCGA-IB-AAUU-01A.tsv     tumor\n",
       "146 TCGA-IB-AAUV-01A TCGA-IB-AAUV-01A.tsv     tumor\n",
       "147 TCGA-IB-AAUW-01A TCGA-IB-AAUW-01A.tsv     tumor\n",
       "148 TCGA-L1-A7W4-01A TCGA-L1-A7W4-01A.tsv     tumor\n",
       "149 TCGA-LB-A7SX-01A TCGA-LB-A7SX-01A.tsv     tumor\n",
       "150 TCGA-LB-A8F3-01A TCGA-LB-A8F3-01A.tsv     tumor\n",
       "151 TCGA-LB-A9Q5-01A TCGA-LB-A9Q5-01A.tsv     tumor\n",
       "152 TCGA-M8-A5N4-01A TCGA-M8-A5N4-01A.tsv     tumor\n",
       "153 TCGA-OE-A75W-01A TCGA-OE-A75W-01A.tsv     tumor\n",
       "154 TCGA-PZ-A5RE-01A TCGA-PZ-A5RE-01A.tsv     tumor\n",
       "155 TCGA-Q3-A5QY-01A TCGA-Q3-A5QY-01A.tsv     tumor\n",
       "156 TCGA-Q3-AA2A-01A TCGA-Q3-AA2A-01A.tsv     tumor\n",
       "157 TCGA-RB-A7B8-01A TCGA-RB-A7B8-01A.tsv     tumor\n",
       "158 TCGA-RB-AA9M-01A TCGA-RB-AA9M-01A.tsv     tumor\n",
       "159 TCGA-RL-AAAS-01A TCGA-RL-AAAS-01A.tsv     tumor\n",
       "160 TCGA-S4-A8RM-01A TCGA-S4-A8RM-01A.tsv     tumor\n",
       "161 TCGA-S4-A8RO-01A TCGA-S4-A8RO-01A.tsv     tumor\n",
       "162 TCGA-S4-A8RP-01A TCGA-S4-A8RP-01A.tsv     tumor\n",
       "163 TCGA-US-A774-01A TCGA-US-A774-01A.tsv     tumor\n",
       "164 TCGA-US-A776-01A TCGA-US-A776-01A.tsv     tumor\n",
       "165 TCGA-US-A779-01A TCGA-US-A779-01A.tsv     tumor\n",
       "166 TCGA-US-A77E-01A TCGA-US-A77E-01A.tsv     tumor\n",
       "167 TCGA-US-A77G-01A TCGA-US-A77G-01A.tsv     tumor\n",
       "168 TCGA-US-A77J-01A TCGA-US-A77J-01A.tsv     tumor\n",
       "169 TCGA-XD-AAUG-01A TCGA-XD-AAUG-01A.tsv     tumor\n",
       "170 TCGA-XD-AAUH-01A TCGA-XD-AAUH-01A.tsv     tumor\n",
       "171 TCGA-XD-AAUI-01A TCGA-XD-AAUI-01A.tsv     tumor\n",
       "172 TCGA-XD-AAUL-01A TCGA-XD-AAUL-01A.tsv     tumor\n",
       "173 TCGA-XN-A8T3-01A TCGA-XN-A8T3-01A.tsv     tumor\n",
       "174 TCGA-XN-A8T5-01A TCGA-XN-A8T5-01A.tsv     tumor\n",
       "175 TCGA-YB-A89D-01A TCGA-YB-A89D-01A.tsv     tumor\n",
       "176 TCGA-YH-A8SY-01A TCGA-YH-A8SY-01A.tsv     tumor\n",
       "177 TCGA-YY-A8LH-01A TCGA-YY-A8LH-01A.tsv     tumor\n",
       "178 TCGA-Z5-AAPL-01A TCGA-Z5-AAPL-01A.tsv     tumor\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# Asssign the directory of the rna samples\n",
    "rna_data_directory <- 'data/tumor_samples'\n",
    "\n",
    "# Build the sampleTable based on the data files we have\n",
    "sampleFiles <- list.files(rna_data_directory)\n",
    "sampleNames <- gsub(\".tsv\", \"\", sampleFiles)\n",
    "sampleCondition <- c(rep(\"tumor\", 178))\n",
    "sampleTable <- data.frame(sampleName = sampleNames,\n",
    "                          fileName = sampleFiles,\n",
    "                          condition = sampleCondition)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Make a DESeq dataset using the sample table and directory to the datafiles\n",
    "# The design is set to 1 to indicate that there are no factors dividing these groups\n",
    "# i.e. they are all tumor samples, see: https://support.bioconductor.org/p/64122/\n",
    "dds <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,\n",
    "                                       directory = rna_data_directory,\n",
    "                                       design= ~ 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: rlog() may take a long time with 50 or more samples,\n",
      "vst() is a much faster transformation\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# http://people.duke.edu/~ccc14/duke-hts-2017/Statistics/08032017/DESeq2-Notebook-introduction.html\n",
    "# Tranform raw counts to rlog\n",
    "# Transformation takes several hours, will load R object with results already if already exists \n",
    "\n",
    "if (!file.exists(\"transformed_dds.rds\")) {\n",
    "    \n",
    "        transformed_dds <- rlog(dds, blind=TRUE)\n",
    "        saveRDS(transformed, file = \"transformed_dds.rds\")\n",
    "    \n",
    "    } else {\n",
    "    \n",
    "        transformed_dds <- readRDS(file = \"transformed_dds.rds\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Get the summarized experiement object, which is a matrix of the transformed counts\n",
    "# Save this matrix to the temp folder\n",
    "\n",
    "transformed_dds_matrix <- assay(transformed_dds)\n",
    "write.csv(transformed_dds_matrix, \"temp/transfored_dds_matrix.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the transforemed_dds_matrix with pandas\n",
    "\n",
    "transformed_dds_matrix = pd.read_csv(\"temp/transfored_dds_matrix.csv\")\n",
    "transformed_dds_matrix = transformed_dds_matrix.rename(index=str, columns={\"Unnamed: 0\": \"Ensembl_ID\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate the 75th percentile of each row\n",
    "\n",
    "# Need to slice only the columns for samples\n",
    "sample_columns = transformed_dds_matrix.columns.values[1:]\n",
    "transformed_dds_matrix['75th_percentile'] = transformed_dds_matrix[sample_columns].apply(lambda x: np.percentile(x, 75), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate the median for each row \n",
    "\n",
    "transformed_dds_matrix['median'] = transformed_dds_matrix[sample_columns].apply(lambda x: np.median(x), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download \"merge_clinical\" clinical data from firebrowse http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/PAAD/20160128/gdac.broadinstitute.org_PAAD.Merge_Clinical.Level_1.2016012800.0.0.tar.gz\n",
    "# Extract and place the \"PAAD.merged_only_clinical_clin_format.txt\" file in the \"data/clinical\" folder\n",
    "# Read the Clinical file, in this case I transposed it to keep the clinical feature title as column name\n",
    "clinical_dir = 'data/clinical/PAAD.merged_only_clinical_clin_format.txt'\n",
    "clinical_df = pd.read_csv(clinical_dir, sep='\\t', index_col='V1').T\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>V1</th>\n",
       "      <th>admin.batch_number</th>\n",
       "      <th>admin.bcr</th>\n",
       "      <th>admin.day_of_dcc_upload</th>\n",
       "      <th>admin.disease_code</th>\n",
       "      <th>admin.file_uuid</th>\n",
       "      <th>admin.month_of_dcc_upload</th>\n",
       "      <th>admin.patient_withdrawal.withdrawn</th>\n",
       "      <th>admin.project_code</th>\n",
       "      <th>admin.year_of_dcc_upload</th>\n",
       "      <th>patient.additional_studies</th>\n",
       "      <th>...</th>\n",
       "      <th>patient.tissue_prospective_collection_indicator</th>\n",
       "      <th>patient.tissue_retrospective_collection_indicator</th>\n",
       "      <th>patient.tissue_source_site</th>\n",
       "      <th>patient.tobacco_smoking_history</th>\n",
       "      <th>patient.tumor_tissue_site</th>\n",
       "      <th>patient.tumor_type</th>\n",
       "      <th>patient.vital_status</th>\n",
       "      <th>patient.year_of_form_completion</th>\n",
       "      <th>patient.year_of_initial_pathologic_diagnosis</th>\n",
       "      <th>patient.year_of_tobacco_smoking_onset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>V2</th>\n",
       "      <td>424.43.0</td>\n",
       "      <td>nationwide children's hospital</td>\n",
       "      <td>21</td>\n",
       "      <td>paad</td>\n",
       "      <td>cda0d295-ea13-4065-9e5d-8177e827a8f6</td>\n",
       "      <td>1</td>\n",
       "      <td>false</td>\n",
       "      <td>tcga</td>\n",
       "      <td>2016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>2j</td>\n",
       "      <td>3</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>primary</td>\n",
       "      <td>dead</td>\n",
       "      <td>2014</td>\n",
       "      <td>2012</td>\n",
       "      <td>1962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V3</th>\n",
       "      <td>424.43.0</td>\n",
       "      <td>nationwide children's hospital</td>\n",
       "      <td>21</td>\n",
       "      <td>paad</td>\n",
       "      <td>d25a44b7-47b1-4a1b-ae08-7bd9aceab9c8</td>\n",
       "      <td>1</td>\n",
       "      <td>false</td>\n",
       "      <td>tcga</td>\n",
       "      <td>2016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>2j</td>\n",
       "      <td>5</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>primary</td>\n",
       "      <td>alive</td>\n",
       "      <td>2014</td>\n",
       "      <td>2012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V4</th>\n",
       "      <td>424.43.0</td>\n",
       "      <td>nationwide children's hospital</td>\n",
       "      <td>21</td>\n",
       "      <td>paad</td>\n",
       "      <td>6f72d859-d6af-4a35-8836-2e086684d90f</td>\n",
       "      <td>1</td>\n",
       "      <td>false</td>\n",
       "      <td>tcga</td>\n",
       "      <td>2016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>2j</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>primary</td>\n",
       "      <td>dead</td>\n",
       "      <td>2014</td>\n",
       "      <td>2012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V5</th>\n",
       "      <td>424.43.0</td>\n",
       "      <td>nationwide children's hospital</td>\n",
       "      <td>21</td>\n",
       "      <td>paad</td>\n",
       "      <td>ca534fd7-dad2-4bd8-9017-391bb314c4a8</td>\n",
       "      <td>1</td>\n",
       "      <td>false</td>\n",
       "      <td>tcga</td>\n",
       "      <td>2016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>2j</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>primary</td>\n",
       "      <td>alive</td>\n",
       "      <td>2014</td>\n",
       "      <td>2012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V6</th>\n",
       "      <td>424.43.0</td>\n",
       "      <td>nationwide children's hospital</td>\n",
       "      <td>21</td>\n",
       "      <td>paad</td>\n",
       "      <td>650295d6-db83-4fa3-909e-4640a77e6274</td>\n",
       "      <td>1</td>\n",
       "      <td>false</td>\n",
       "      <td>tcga</td>\n",
       "      <td>2016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>2j</td>\n",
       "      <td>1</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>primary</td>\n",
       "      <td>dead</td>\n",
       "      <td>2014</td>\n",
       "      <td>2012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 440 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "V1 admin.batch_number                       admin.bcr admin.day_of_dcc_upload  \\\n",
       "V2           424.43.0  nationwide children's hospital                      21   \n",
       "V3           424.43.0  nationwide children's hospital                      21   \n",
       "V4           424.43.0  nationwide children's hospital                      21   \n",
       "V5           424.43.0  nationwide children's hospital                      21   \n",
       "V6           424.43.0  nationwide children's hospital                      21   \n",
       "\n",
       "V1 admin.disease_code                       admin.file_uuid  \\\n",
       "V2               paad  cda0d295-ea13-4065-9e5d-8177e827a8f6   \n",
       "V3               paad  d25a44b7-47b1-4a1b-ae08-7bd9aceab9c8   \n",
       "V4               paad  6f72d859-d6af-4a35-8836-2e086684d90f   \n",
       "V5               paad  ca534fd7-dad2-4bd8-9017-391bb314c4a8   \n",
       "V6               paad  650295d6-db83-4fa3-909e-4640a77e6274   \n",
       "\n",
       "V1 admin.month_of_dcc_upload admin.patient_withdrawal.withdrawn  \\\n",
       "V2                         1                              false   \n",
       "V3                         1                              false   \n",
       "V4                         1                              false   \n",
       "V5                         1                              false   \n",
       "V6                         1                              false   \n",
       "\n",
       "V1 admin.project_code admin.year_of_dcc_upload patient.additional_studies  \\\n",
       "V2               tcga                     2016                        NaN   \n",
       "V3               tcga                     2016                        NaN   \n",
       "V4               tcga                     2016                        NaN   \n",
       "V5               tcga                     2016                        NaN   \n",
       "V6               tcga                     2016                        NaN   \n",
       "\n",
       "V1  ... patient.tissue_prospective_collection_indicator  \\\n",
       "V2  ...                                              no   \n",
       "V3  ...                                              no   \n",
       "V4  ...                                              no   \n",
       "V5  ...                                              no   \n",
       "V6  ...                                              no   \n",
       "\n",
       "V1 patient.tissue_retrospective_collection_indicator  \\\n",
       "V2                                               yes   \n",
       "V3                                               yes   \n",
       "V4                                               yes   \n",
       "V5                                               yes   \n",
       "V6                                               yes   \n",
       "\n",
       "V1 patient.tissue_source_site patient.tobacco_smoking_history  \\\n",
       "V2                         2j                               3   \n",
       "V3                         2j                               5   \n",
       "V4                         2j                             NaN   \n",
       "V5                         2j                             NaN   \n",
       "V6                         2j                               1   \n",
       "\n",
       "V1 patient.tumor_tissue_site patient.tumor_type patient.vital_status  \\\n",
       "V2                  pancreas            primary                 dead   \n",
       "V3                  pancreas            primary                alive   \n",
       "V4                  pancreas            primary                 dead   \n",
       "V5                  pancreas            primary                alive   \n",
       "V6                  pancreas            primary                 dead   \n",
       "\n",
       "V1 patient.year_of_form_completion  \\\n",
       "V2                            2014   \n",
       "V3                            2014   \n",
       "V4                            2014   \n",
       "V5                            2014   \n",
       "V6                            2014   \n",
       "\n",
       "V1 patient.year_of_initial_pathologic_diagnosis  \\\n",
       "V2                                         2012   \n",
       "V3                                         2012   \n",
       "V4                                         2012   \n",
       "V5                                         2012   \n",
       "V6                                         2012   \n",
       "\n",
       "V1 patient.year_of_tobacco_smoking_onset  \n",
       "V2                                  1962  \n",
       "V3                                   NaN  \n",
       "V4                                   NaN  \n",
       "V5                                   NaN  \n",
       "V6                                   NaN  \n",
       "\n",
       "[5 rows x 440 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Visualize the head of the file\n",
    "clinical_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert patient barcodes to uppercase\n",
    "clinical_df['patient.bcr_patient_barcode'] = clinical_df['patient.bcr_patient_barcode'].str.upper()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Days to new tumor event after initial treatment are listed in several columns.\n",
    "# If there are multiple records for this, we need to take the first value (earliest recorded value)\n",
    "\n",
    "new_tumor_df = pd.DataFrame()\n",
    "new_tumor_df['patient.bcr_patient_barcode']  = clinical_df['patient.bcr_patient_barcode']\n",
    "for column in clinical_df:\n",
    "    if 'days_to_new_tumor_event_after_initial_treatment' in column:\n",
    "        new_tumor_df[column] = clinical_df[column]\n",
    "        new_tumor_df[column] = new_tumor_df[column].astype(float)\n",
    "        new_tumor_df['new_tumor_days'] = new_tumor_df.min(axis=1)\n",
    "        \n",
    "new_tumor_df = new_tumor_df[['patient.bcr_patient_barcode', 'new_tumor_days']]       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Similarly get the earliest recording of days to death\n",
    "# This should be unambiguous, but we take the max anyway just to pick one value if there are multiple entries\n",
    "\n",
    "death_df = pd.DataFrame()\n",
    "death_df['patient.bcr_patient_barcode']  = clinical_df['patient.bcr_patient_barcode']\n",
    "for column in clinical_df:\n",
    "    if 'days_to_death' in column:\n",
    "        death_df[column] = clinical_df[column]\n",
    "        death_df[column] = death_df[column].astype(float)\n",
    "        death_df['death_days'] = death_df.max(axis=1)\n",
    "\n",
    "death_df = death_df[['patient.bcr_patient_barcode', 'death_days']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Similarly get the most recent number for \"days to last followup\"\n",
    "# Days_to_last_followup is the time interval from the date of last followup to the date of initial pathologic diagnosis\n",
    "# This means the most recent followup would be the largest number\n",
    "\n",
    "followup_df = pd.DataFrame()\n",
    "followup_df['patient.bcr_patient_barcode']  = clinical_df['patient.bcr_patient_barcode']\n",
    "for column in clinical_df:\n",
    "    if 'days_to_last_followup' in column:\n",
    "        followup_df[column] = clinical_df[column]\n",
    "        followup_df[column] = followup_df[column].astype(float)\n",
    "        followup_df['followup_days'] = followup_df.max(axis=1)\n",
    "\n",
    "followup_df = followup_df[['patient.bcr_patient_barcode', 'followup_days']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge new_tumor_df, death_df, followup_df\n",
    "df_list = [new_tumor_df, death_df, followup_df]\n",
    "all_clinical_df = pd.DataFrame()\n",
    "all_clinical_df['patient.bcr_patient_barcode'] = clinical_df['patient.bcr_patient_barcode']\n",
    "for df in df_list:\n",
    "    all_clinical_df = all_clinical_df.merge(df, on='patient.bcr_patient_barcode')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to create a column for death that takes into account samples that need to be censored (patients that did not die)\n",
    "# If the patient did die, the time to death will be recorded in this column\n",
    "# If the patient did not die, the number of days since last followup will be recorded in this column\n",
    "\n",
    "all_clinical_df['new_death']= np.where(all_clinical_df['death_days'].isnull(),all_clinical_df['followup_days'] , all_clinical_df['death_days'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to create a column for time to new tumor that takes into account censored patients\n",
    "# If the patient has a non-null new_tumor_days value, this value will be recorded in this column\n",
    "# If the patient did not develop a new tumor, then the days since last followup will be used in this column \n",
    "all_clinical_df['new_time'] = np.where(all_clinical_df['new_tumor_days'].isnull(), all_clinical_df['followup_days'], all_clinical_df['new_tumor_days'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/brett/.local/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "# Translate the alive and dead strings in the patient.vital_status column to 1 for dead and 0 for alive\n",
    "\n",
    "alive_dead_df = clinical_df[['patient.bcr_patient_barcode', 'patient.vital_status']]\n",
    "alive_dead_df['death_event'] = np.where(alive_dead_df['patient.vital_status'] == 'dead', 1, 0)\n",
    "alive_dead_df = alive_dead_df[['patient.bcr_patient_barcode', 'death_event']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge the alive/dead data with the all_clincal_df data\n",
    "\n",
    "all_clinical_df = all_clinical_df.merge(alive_dead_df, on='patient.bcr_patient_barcode')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000105976.13\n"
     ]
    }
   ],
   "source": [
    "# Function can be used to find a gene from an ensembl root\n",
    "def find_gene_id(id):\n",
    "    for item in transformed_dds_matrix['Ensembl_ID']:\n",
    "        if id in item:\n",
    "            return item\n",
    "            \n",
    "# The root ensembl id for MET is ENSG00000105976\n",
    "\n",
    "print(find_gene_id('ENSG00000105976'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl_ID</th>\n",
       "      <th>TCGA-2J-AAB1-01A.tsv</th>\n",
       "      <th>TCGA-2J-AAB4-01A.tsv</th>\n",
       "      <th>TCGA-2J-AAB6-01A.tsv</th>\n",
       "      <th>TCGA-2J-AAB8-01A.tsv</th>\n",
       "      <th>TCGA-2J-AAB9-01A.tsv</th>\n",
       "      <th>TCGA-2J-AABA-01A.tsv</th>\n",
       "      <th>TCGA-2J-AABE-01A.tsv</th>\n",
       "      <th>TCGA-2J-AABF-01A.tsv</th>\n",
       "      <th>TCGA-2J-AABH-01A.tsv</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-XD-AAUI-01A.tsv</th>\n",
       "      <th>TCGA-XD-AAUL-01A.tsv</th>\n",
       "      <th>TCGA-XN-A8T3-01A.tsv</th>\n",
       "      <th>TCGA-XN-A8T5-01A.tsv</th>\n",
       "      <th>TCGA-YB-A89D-01A.tsv</th>\n",
       "      <th>TCGA-YH-A8SY-01A.tsv</th>\n",
       "      <th>TCGA-YY-A8LH-01A.tsv</th>\n",
       "      <th>TCGA-Z5-AAPL-01A.tsv</th>\n",
       "      <th>75th_percentile</th>\n",
       "      <th>median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3159</th>\n",
       "      <td>ENSG00000105976.13</td>\n",
       "      <td>12.579738</td>\n",
       "      <td>12.391897</td>\n",
       "      <td>13.724667</td>\n",
       "      <td>12.40901</td>\n",
       "      <td>12.661981</td>\n",
       "      <td>12.556881</td>\n",
       "      <td>12.467062</td>\n",
       "      <td>12.655133</td>\n",
       "      <td>12.918456</td>\n",
       "      <td>...</td>\n",
       "      <td>12.686107</td>\n",
       "      <td>12.854646</td>\n",
       "      <td>11.91496</td>\n",
       "      <td>12.755337</td>\n",
       "      <td>12.567001</td>\n",
       "      <td>12.261489</td>\n",
       "      <td>12.421034</td>\n",
       "      <td>11.478104</td>\n",
       "      <td>12.958448</td>\n",
       "      <td>12.531897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 181 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Ensembl_ID  TCGA-2J-AAB1-01A.tsv  TCGA-2J-AAB4-01A.tsv  \\\n",
       "3159  ENSG00000105976.13             12.579738             12.391897   \n",
       "\n",
       "      TCGA-2J-AAB6-01A.tsv  TCGA-2J-AAB8-01A.tsv  TCGA-2J-AAB9-01A.tsv  \\\n",
       "3159             13.724667              12.40901             12.661981   \n",
       "\n",
       "      TCGA-2J-AABA-01A.tsv  TCGA-2J-AABE-01A.tsv  TCGA-2J-AABF-01A.tsv  \\\n",
       "3159             12.556881             12.467062             12.655133   \n",
       "\n",
       "      TCGA-2J-AABH-01A.tsv  ...  TCGA-XD-AAUI-01A.tsv  TCGA-XD-AAUL-01A.tsv  \\\n",
       "3159             12.918456  ...             12.686107             12.854646   \n",
       "\n",
       "      TCGA-XN-A8T3-01A.tsv  TCGA-XN-A8T5-01A.tsv  TCGA-YB-A89D-01A.tsv  \\\n",
       "3159              11.91496             12.755337             12.567001   \n",
       "\n",
       "      TCGA-YH-A8SY-01A.tsv  TCGA-YY-A8LH-01A.tsv  TCGA-Z5-AAPL-01A.tsv  \\\n",
       "3159             12.261489             12.421034             11.478104   \n",
       "\n",
       "      75th_percentile     median  \n",
       "3159        12.958448  12.531897  \n",
       "\n",
       "[1 rows x 181 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Select row from transformed_dds_matrix with our gene of interest\n",
    "gene_of_interest = find_gene_id('ENSG00000105976')\n",
    "transformed_dds_matrix[transformed_dds_matrix['Ensembl_ID'] == gene_of_interest]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient.bcr_patient_barcode</th>\n",
       "      <th>new_tumor_days</th>\n",
       "      <th>death_days</th>\n",
       "      <th>followup_days</th>\n",
       "      <th>new_death</th>\n",
       "      <th>new_time</th>\n",
       "      <th>death_event</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2J-AAB1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>66.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>66.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2J-AAB4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>729.0</td>\n",
       "      <td>729.0</td>\n",
       "      <td>729.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-2J-AAB6</td>\n",
       "      <td>169.0</td>\n",
       "      <td>293.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>293.0</td>\n",
       "      <td>169.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-2J-AAB8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-2J-AAB9</td>\n",
       "      <td>521.0</td>\n",
       "      <td>627.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>627.0</td>\n",
       "      <td>521.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TCGA-2J-AABA</td>\n",
       "      <td>160.0</td>\n",
       "      <td>607.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>607.0</td>\n",
       "      <td>160.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TCGA-2J-AABE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>676.0</td>\n",
       "      <td>676.0</td>\n",
       "      <td>676.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TCGA-2J-AABF</td>\n",
       "      <td>443.0</td>\n",
       "      <td>691.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>691.0</td>\n",
       "      <td>443.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TCGA-2J-AABH</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>1287.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TCGA-2J-AABI</td>\n",
       "      <td>728.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>969.0</td>\n",
       "      <td>969.0</td>\n",
       "      <td>728.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TCGA-2J-AABK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>484.0</td>\n",
       "      <td>484.0</td>\n",
       "      <td>484.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TCGA-2J-AABO</td>\n",
       "      <td>439.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>440.0</td>\n",
       "      <td>440.0</td>\n",
       "      <td>439.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TCGA-2J-AABP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>463.0</td>\n",
       "      <td>463.0</td>\n",
       "      <td>463.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TCGA-2J-AABR</td>\n",
       "      <td>327.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>438.0</td>\n",
       "      <td>438.0</td>\n",
       "      <td>327.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TCGA-2J-AABT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>319.0</td>\n",
       "      <td>319.0</td>\n",
       "      <td>319.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-2J-AABU</td>\n",
       "      <td>154.0</td>\n",
       "      <td>277.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>277.0</td>\n",
       "      <td>154.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TCGA-2J-AABV</td>\n",
       "      <td>126.0</td>\n",
       "      <td>652.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>652.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TCGA-2L-AAQA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>143.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>143.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TCGA-2L-AAQE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>684.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>684.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TCGA-2L-AAQI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>103.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>103.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TCGA-2L-AAQJ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>394.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>394.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TCGA-2L-AAQL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>292.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>292.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TCGA-2L-AAQM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1383.0</td>\n",
       "      <td>1383.0</td>\n",
       "      <td>1383.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TCGA-3A-A9I5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1794.0</td>\n",
       "      <td>1794.0</td>\n",
       "      <td>1794.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TCGA-3A-A9I7</td>\n",
       "      <td>1309.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1323.0</td>\n",
       "      <td>1323.0</td>\n",
       "      <td>1309.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>TCGA-3A-A9I9</td>\n",
       "      <td>513.0</td>\n",
       "      <td>634.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>634.0</td>\n",
       "      <td>513.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>TCGA-3A-A9IB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>224.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>224.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TCGA-3A-A9IC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>738.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>738.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TCGA-3A-A9IH</td>\n",
       "      <td>763.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1021.0</td>\n",
       "      <td>1021.0</td>\n",
       "      <td>763.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TCGA-3A-A9IJ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1854.0</td>\n",
       "      <td>1854.0</td>\n",
       "      <td>1854.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>TCGA-LB-A7SX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>393.0</td>\n",
       "      <td>127.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>TCGA-LB-A8F3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>379.0</td>\n",
       "      <td>379.0</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>TCGA-LB-A9Q5</td>\n",
       "      <td>290.0</td>\n",
       "      <td>313.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>313.0</td>\n",
       "      <td>290.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>TCGA-M8-A5N4</td>\n",
       "      <td>416.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>584.0</td>\n",
       "      <td>584.0</td>\n",
       "      <td>416.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>TCGA-OE-A75W</td>\n",
       "      <td>146.0</td>\n",
       "      <td>267.0</td>\n",
       "      <td>110.0</td>\n",
       "      <td>267.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>TCGA-PZ-A5RE</td>\n",
       "      <td>393.0</td>\n",
       "      <td>470.0</td>\n",
       "      <td>247.0</td>\n",
       "      <td>470.0</td>\n",
       "      <td>393.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>TCGA-Q3-A5QY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>416.0</td>\n",
       "      <td>416.0</td>\n",
       "      <td>416.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>TCGA-Q3-AA2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>95.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>95.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>TCGA-RB-A7B8</td>\n",
       "      <td>449.0</td>\n",
       "      <td>466.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>466.0</td>\n",
       "      <td>449.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>TCGA-RB-AA9M</td>\n",
       "      <td>258.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>286.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>258.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>TCGA-RL-AAAS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>TCGA-S4-A8RM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>737.0</td>\n",
       "      <td>737.0</td>\n",
       "      <td>737.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>TCGA-S4-A8RO</td>\n",
       "      <td>291.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>525.0</td>\n",
       "      <td>525.0</td>\n",
       "      <td>291.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>TCGA-S4-A8RP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>702.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>702.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>TCGA-US-A774</td>\n",
       "      <td>366.0</td>\n",
       "      <td>695.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>695.0</td>\n",
       "      <td>366.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>TCGA-US-A776</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>1216.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>TCGA-US-A779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>511.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>511.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>TCGA-US-A77E</td>\n",
       "      <td>166.0</td>\n",
       "      <td>430.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>430.0</td>\n",
       "      <td>166.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>TCGA-US-A77G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>TCGA-US-A77J</td>\n",
       "      <td>NaN</td>\n",
       "      <td>568.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>568.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>TCGA-XD-AAUG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>420.0</td>\n",
       "      <td>420.0</td>\n",
       "      <td>420.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>TCGA-XD-AAUH</td>\n",
       "      <td>87.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>395.0</td>\n",
       "      <td>395.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>TCGA-XD-AAUI</td>\n",
       "      <td>175.0</td>\n",
       "      <td>366.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>366.0</td>\n",
       "      <td>175.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>TCGA-XD-AAUL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>498.0</td>\n",
       "      <td>498.0</td>\n",
       "      <td>498.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>TCGA-XN-A8T3</td>\n",
       "      <td>461.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>951.0</td>\n",
       "      <td>951.0</td>\n",
       "      <td>461.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>TCGA-XN-A8T5</td>\n",
       "      <td>242.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>720.0</td>\n",
       "      <td>720.0</td>\n",
       "      <td>242.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>TCGA-YB-A89D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>350.0</td>\n",
       "      <td>350.0</td>\n",
       "      <td>350.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>TCGA-YH-A8SY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>388.0</td>\n",
       "      <td>388.0</td>\n",
       "      <td>388.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>TCGA-YY-A8LH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>TCGA-Z5-AAPL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>467.0</td>\n",
       "      <td>467.0</td>\n",
       "      <td>467.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    patient.bcr_patient_barcode  new_tumor_days  death_days  followup_days  \\\n",
       "0                  TCGA-2J-AAB1             NaN        66.0            NaN   \n",
       "1                  TCGA-2J-AAB4             NaN         NaN          729.0   \n",
       "2                  TCGA-2J-AAB6           169.0       293.0            NaN   \n",
       "3                  TCGA-2J-AAB8             NaN         NaN           80.0   \n",
       "4                  TCGA-2J-AAB9           521.0       627.0            NaN   \n",
       "5                  TCGA-2J-AABA           160.0       607.0            NaN   \n",
       "6                  TCGA-2J-AABE             NaN         NaN          676.0   \n",
       "7                  TCGA-2J-AABF           443.0       691.0            NaN   \n",
       "8                  TCGA-2J-AABH          1287.0         NaN         1287.0   \n",
       "9                  TCGA-2J-AABI           728.0         NaN          969.0   \n",
       "10                 TCGA-2J-AABK             NaN         NaN          484.0   \n",
       "11                 TCGA-2J-AABO           439.0         NaN          440.0   \n",
       "12                 TCGA-2J-AABP             NaN         NaN          463.0   \n",
       "13                 TCGA-2J-AABR           327.0         NaN          438.0   \n",
       "14                 TCGA-2J-AABT             NaN         NaN          319.0   \n",
       "15                 TCGA-2J-AABU           154.0       277.0            NaN   \n",
       "16                 TCGA-2J-AABV           126.0       652.0            NaN   \n",
       "17                 TCGA-2L-AAQA             NaN       143.0            NaN   \n",
       "18                 TCGA-2L-AAQE             NaN       684.0            NaN   \n",
       "19                 TCGA-2L-AAQI             NaN       103.0            NaN   \n",
       "20                 TCGA-2L-AAQJ             NaN       394.0            NaN   \n",
       "21                 TCGA-2L-AAQL             NaN       292.0            NaN   \n",
       "22                 TCGA-2L-AAQM             NaN         NaN         1383.0   \n",
       "23                 TCGA-3A-A9I5             NaN         NaN         1794.0   \n",
       "24                 TCGA-3A-A9I7          1309.0         NaN         1323.0   \n",
       "25                 TCGA-3A-A9I9           513.0       634.0            NaN   \n",
       "26                 TCGA-3A-A9IB             NaN       224.0            NaN   \n",
       "27                 TCGA-3A-A9IC             NaN       738.0            NaN   \n",
       "28                 TCGA-3A-A9IH           763.0         NaN         1021.0   \n",
       "29                 TCGA-3A-A9IJ             NaN         NaN         1854.0   \n",
       "..                          ...             ...         ...            ...   \n",
       "155                TCGA-LB-A7SX             NaN       393.0          127.0   \n",
       "156                TCGA-LB-A8F3             NaN         NaN          379.0   \n",
       "157                TCGA-LB-A9Q5           290.0       313.0          155.0   \n",
       "158                TCGA-M8-A5N4           416.0         NaN          584.0   \n",
       "159                TCGA-OE-A75W           146.0       267.0          110.0   \n",
       "160                TCGA-PZ-A5RE           393.0       470.0          247.0   \n",
       "161                TCGA-Q3-A5QY             NaN         NaN          416.0   \n",
       "162                TCGA-Q3-AA2A             NaN         NaN           95.0   \n",
       "163                TCGA-RB-A7B8           449.0       466.0           36.0   \n",
       "164                TCGA-RB-AA9M           258.0         NaN          286.0   \n",
       "165                TCGA-RL-AAAS             NaN         NaN            9.0   \n",
       "166                TCGA-S4-A8RM             NaN         NaN          737.0   \n",
       "167                TCGA-S4-A8RO           291.0         NaN          525.0   \n",
       "168                TCGA-S4-A8RP             NaN       702.0            NaN   \n",
       "169                TCGA-US-A774           366.0       695.0            NaN   \n",
       "170                TCGA-US-A776             NaN         NaN         1216.0   \n",
       "171                TCGA-US-A779             NaN       511.0            NaN   \n",
       "172                TCGA-US-A77E           166.0       430.0            NaN   \n",
       "173                TCGA-US-A77G             NaN        12.0            NaN   \n",
       "174                TCGA-US-A77J             NaN       568.0            NaN   \n",
       "175                TCGA-XD-AAUG             NaN         NaN          420.0   \n",
       "176                TCGA-XD-AAUH            87.0         NaN          395.0   \n",
       "177                TCGA-XD-AAUI           175.0       366.0          202.0   \n",
       "178                TCGA-XD-AAUL             NaN         NaN          498.0   \n",
       "179                TCGA-XN-A8T3           461.0         NaN          951.0   \n",
       "180                TCGA-XN-A8T5           242.0         NaN          720.0   \n",
       "181                TCGA-YB-A89D             NaN         NaN          350.0   \n",
       "182                TCGA-YH-A8SY             NaN         NaN          388.0   \n",
       "183                TCGA-YY-A8LH             NaN         NaN         2016.0   \n",
       "184                TCGA-Z5-AAPL             NaN         NaN          467.0   \n",
       "\n",
       "     new_death  new_time  death_event  \n",
       "0         66.0       NaN            1  \n",
       "1        729.0     729.0            0  \n",
       "2        293.0     169.0            1  \n",
       "3         80.0      80.0            0  \n",
       "4        627.0     521.0            1  \n",
       "5        607.0     160.0            1  \n",
       "6        676.0     676.0            0  \n",
       "7        691.0     443.0            1  \n",
       "8       1287.0    1287.0            0  \n",
       "9        969.0     728.0            0  \n",
       "10       484.0     484.0            0  \n",
       "11       440.0     439.0            0  \n",
       "12       463.0     463.0            0  \n",
       "13       438.0     327.0            0  \n",
       "14       319.0     319.0            0  \n",
       "15       277.0     154.0            1  \n",
       "16       652.0     126.0            1  \n",
       "17       143.0       NaN            1  \n",
       "18       684.0       NaN            1  \n",
       "19       103.0       NaN            1  \n",
       "20       394.0       NaN            1  \n",
       "21       292.0       NaN            1  \n",
       "22      1383.0    1383.0            0  \n",
       "23      1794.0    1794.0            0  \n",
       "24      1323.0    1309.0            0  \n",
       "25       634.0     513.0            1  \n",
       "26       224.0       NaN            1  \n",
       "27       738.0       NaN            1  \n",
       "28      1021.0     763.0            0  \n",
       "29      1854.0    1854.0            0  \n",
       "..         ...       ...          ...  \n",
       "155      393.0     127.0            0  \n",
       "156      379.0     379.0            0  \n",
       "157      313.0     290.0            0  \n",
       "158      584.0     416.0            0  \n",
       "159      267.0     146.0            0  \n",
       "160      470.0     393.0            0  \n",
       "161      416.0     416.0            0  \n",
       "162       95.0      95.0            0  \n",
       "163      466.0     449.0            0  \n",
       "164      286.0     258.0            0  \n",
       "165        9.0       9.0            0  \n",
       "166      737.0     737.0            0  \n",
       "167      525.0     291.0            0  \n",
       "168      702.0       NaN            1  \n",
       "169      695.0     366.0            1  \n",
       "170     1216.0    1216.0            0  \n",
       "171      511.0       NaN            1  \n",
       "172      430.0     166.0            1  \n",
       "173       12.0       NaN            1  \n",
       "174      568.0       NaN            1  \n",
       "175      420.0     420.0            0  \n",
       "176      395.0      87.0            0  \n",
       "177      366.0     175.0            0  \n",
       "178      498.0     498.0            0  \n",
       "179      951.0     461.0            0  \n",
       "180      720.0     242.0            0  \n",
       "181      350.0     350.0            0  \n",
       "182      388.0     388.0            0  \n",
       "183     2016.0    2016.0            0  \n",
       "184      467.0     467.0            0  \n",
       "\n",
       "[185 rows x 7 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a list \n",
    "all_clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
